vm-setup/roles/v1aX_integration_test/tasks/move.yml

```diff
@@ -95,53 +108,53 @@
 
   # Check for pods & nodes on the target cluster
   - name: Check if pods in running state
-    shell: "kubectl get pods -A -o json | jq -r '.items[].status.phase' | grep -v Running"
+    k8s_info:
+      kind: pods 
+      field_selectors:
+        - status.phase!=Running
+    register: target_running_pods 
     environment:
       KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
-    retries: 150
+    retries: 150 
     delay: 20
-    register: target_running_pods
-    failed_when: >
-      (target_running_pods.stderr != "") or
-      (target_running_pods.rc > 1) or
-      (target_running_pods.stdout != "")
-    until: target_running_pods.stdout == ""
-
+    until: target_running_pods.resources | length == 0
+    failed_when: target_running_pods.resources | length > 0 
+ 
   - name: Pivot objects to target cluster
     shell: "clusterctl move --to-kubeconfig /tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml -n {{ NAMESPACE }} -v 10"
 
   - name: Check if machines become running.
-    shell: |
-        kubectl get machines -n {{ NAMESPACE }} -o json | jq -r '[ .items[]
-        | select (.status.phase == "Running" or .status.phase == "running")
-        | .metadata.name ] | length'
-    register: provisioned_machines
+    k8s_info:
+      api_version: cluster.x-k8s.io/v1alpha3 
+      kind: machines 
+    register: provisioned_machines 
     environment:
       KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
-    retries: 50
+    retries: 50 
     delay: 20
-    until: provisioned_machines.stdout == NUMBER_OF_BMH
-
+    until: provisioned_machines | running | length == NUMBER_OF_BMH
+  
   - name: Check if metal3machines become provisioned.
-    shell: |
-        kubectl get m3m -n {{ NAMESPACE }} -o json | jq -r '[ .items[]
-        | select (.status.ready == true)
-        | .metadata.name ] | length'
+    k8s_info:
+      api_version: infrastructure.cluster.x-k8s.io/v1alpha4
+      kind: metal3machine 
+      namespace: "{{ NAMESPACE }}"
     register: provisioned_m3m_machines
     environment:
       KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
     retries: 10
     delay: 20
-    until: provisioned_m3m_machines.stdout == NUMBER_OF_BMH
+    until: provisioned_m3m_machines | ready | length == NUMBER_OF_BMH    
 
   - name: Check if bmh is in provisioned state
-    shell: |
-        kubectl get bmh -n {{ NAMESPACE }} -o json | jq -r '[ .items[]
-        | select (.status.provisioning.state == "provisioned")
-        | .metadata.name ] | length'
+    k8s_info:
+      api_version: metal3.io/v1alpha1 
+      kind: BareMetalHost 
+      namespace: "{{ NAMESPACE }}"
+    register: provisioned_bmh
     environment:
       KUBECONFIG: "/tmp/kubeconfig-{{ CLUSTER_NAME }}.yaml"
-    register: provisioned_bmh
     retries: 10
     delay: 20
-    until: provisioned_bmh.stdout ==  NUMBER_OF_BMH
+    until: provisioned_bmh | provisioned | length == NUMBER_OF_BMH 
```

Comment:
The LHS is an `int` but the RHS is a `string`. You'll need to cast the RHS to an `int` to do the comparison.